wordpress-mobile/WordPress-iOS
# My Site Dashboard: Refactor next post cells and remove unneeded code

Part of #18308

## Description
- This PR concludes the refactor 🎉 
- Refactored `BlogDashboardNextPostView` into a collection view cell `DashboardEmptyPostsCardCell`
- Added two cells inheriting from `DashboardEmptyPostsCardCell`
- Removed `DashboardPostCardCell` and `PostsCardViewController` completely
- Removed unnecessary code from multiple places
- Handles cases where cards needs to be reloaded to correctly show the current state

_PS: I know the lines changed are a LOT. However, ~500 of those are deleted files 😄_  

## Testing Instructions

Since the refactor is now done, please test that the dashboard behaves as expected.
- Sanity check on the dashboard in general:
    - Quick actions
    - Stats
    - Quick start
    - Loading state (Ghost)
    - Failure card
- A more detailed sanity check on posts cards. Some scenarios to consider:
    - Test drafts, scheduled, create next, create first cards are shown when expected
    - Test draft and scheduled lists are updated normally (Adding or removing posts)
    - Test drafts and scheduled cards are removed when all posts are removed (either by trashing or publishing)
    - Test showing drafts card after the first draft is created
    - Test showing scheduled card after the first scheduled post is created
    - Test switching from create first card to create next card when the first post is published   

## Regression Notes
1. Potential unintended areas of impact
Posts cards

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
